ministryofjustice/analytics-platform-control-panel
# Bugfix/cancel task

## :memo: Summary
This PR relates to https://github.com/ministryofjustice/analytical-platform/issues/5528

NOTE: all changes relate to admin/superuser pages only, they are not user facing. The changes are intended to make it easier for developers to debug when a task fails.

- Add a task detail page with additional information about the task
- Add a field to mark a task as cancelled
- Add a button to cancel a task
- If a task is cancelled, celery will remove it from the task queue


## :mag: What should the reviewer concentrate on?
- Task detail page

## :technologist: How should the reviewer test these changes?
- Because there are differences between how tasks are sent when running locally, the best way to test this will be in the dev environment
- After it has been deployed, [visit the incomplete task list page](https://controlpanel.services.dev.analytical-platform.service.justice.gov.uk/)
- Check the task detail page to check that the extra information displays
- Try cancelling a task

## :books: Documentation status

- [x] Documentation will be added in the future because I want to test the changes in dev first
